### ISSUE TITLE
Infinite thumbnail generation on remote storages

### ISSUE THREAD

**Issue by MythicManiac** [SmileyChris/easy-thumbnails#513]
## Introduction

Today I struggled with a rather peculiar issue, seeing that one of my google cloud buckets was filling up with thumbnails. Here is an example list of the thumbnails that were residing on the bucket: [thunderstore-live-files-filtered.txt](https://github.com/SmileyChris/easy-thumbnails/files/3052117/thunderstore-live-files-filtered.txt).

As to how this kind of an issue can be encountered, here's my current theory:

## Conditions

- Usage of a storage backend that prevents overwriting files, but instead creates new files (that will end up as the duplicates).
- Modify time mismatch between a thumbnail source, and a thumbnail cache, where the thumbnail file is older than the actual source.
  - This I persume to have happened on Google's part on the cloud storage, as there should be no way of the thumbnail coming _before_ the source, given how overwrites are not possible.

## Clarification

Since we know that no files on the cloud storage can be overwritten or modified, we can assume that all file modification times on the storage backend are actually file creation times.

However, it seems that due to timing inconsistencies in networked storages, it is possible for the thumbnail object to receive a timestamp _earlier_ in time than the thumbnail source itself.

This will in effect cause easy_thumbnails to attempt updating the thumbnail every time it is needed, but being unable to overwrite the old thumbnail, resulting in massive amounts of duplicate thumbnails on the storage backend.

## Suggested actions

I see at least two ways of approaching this issue, one more simple than the another:

### Add an option to disable modification date checking

This issue is only relevant in environments where updating files is not possible, and as such all thumbnails can be assumed to be up to date even if their timestamps don't match. The relevant check can be found from [files.py#L614](https://github.com/SmileyChris/easy-thumbnails/blob/master/easy_thumbnails/files.py#L614)

### Add thumbnail update support for environments where overwriting files is not possible

This would require some architectural changes, but work just as well. Right now the issue is that the newly generated, "up to date" -thumbnails end up with a mismatching filename. If some other method of matching the latest thumbnail would be used in place of the file name, then it could be possible to properly match the correct thumbnail in cases like these as well.

**Comment by MythicManiac** [comment]
TL;DR - if thumbnails are generated 1 minute into the past on a write-once storage, easy_thumbnails dies.

**Comment by MythicManiac** [comment]
I ended up monkeypatching this for myself as a temporary solution here: https://github.com/risk-of-thunder/Thunderstore/commit/ba83baf2c15b13f9000f346b30bf03a0cc13fbfa
